adc/ade7913/flexpwm.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX FlexPWM
+ *
+ * Copyright 2021-2022 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef FLEXPWM_H
+#define FLEXPWM_H
+
+#include <stdint.h>
+
+
+struct flexpwm_submodule_regs_s {
```

Comment:
is this needed to be visible externally or should be moved to implementation file? (doesn't matter now as it's internal to ade driver, I'm just worried somebody will move it in the future ;))

(but yeah, I much more prefer struct-based peripherial access to enum-based, especially You can view the whole peripherial status under `gdb` that way in a sensible way)

Reply:
Done, I moved those structures to the implementation (c file).